### PR TITLE
Add canvas analytics

### PR DESCRIPTION
**Pull request by huwshimi**
Fixes: #3957.

### CODE REVIEW

File: gui/src/app/init/topology/relation.js [hunk 1]
```diff
@@ -31,9 +31,10 @@ const RelationMenu = require('../../components/relation-menu/relation-menu');
  * @class RelationModule
  */
 class RelationModule {
-  constructor() {
+  constructor(options={}) {
```
Comment (hatched) [inline gui/src/app/init/topology/relation.js]: driveby fix?

File: gui/src/app/init/topology/relation.js [hunk 2]
```diff
@@ -648,6 +650,7 @@ class RelationModule {
     // If we landed on a rect, add relation, otherwise, cancel.
     if (rect) {
       self.ambiguousAddRelationCheck(endpoint, self, rect);
+      this.analytics.sendEvent(self.analytics.ADD);
```
Comment (hatched) [inline gui/src/app/init/topology/relation.js]: should `this` be `self`? 

doesn't this need a label or a category?

File: gui/src/app/init/topology/relation.js (hunk 1 above)
Comment (huwshimi) [reply gui/src/app/init/topology/relation.js]: This is require to pass in the analytics instance below (`this.analytics = options.analytics.addCategory('Relation');`).

File: gui/src/app/init/topology/relation.js (hunk 2 above)
Comment (huwshimi) [reply gui/src/app/init/topology/relation.js]: This will result in `Canvas > Relation` action: Add. Do you think it needs something else?

Comment (hatched) [reply gui/src/app/init/topology/relation.js]: Nope that will be fine
